gerrypy/scripts/fish_scales.py

```diff
@@ -0,0 +1,286 @@
+import networkx as nx
+
+
+def fill_graph(request):
+    """Build global graph from tract and edge databases."""
+    graph = nx.Graph()
+    tracts = request.dbsession.query(Tract).all()  # get all tracts from db
+    edges = request.dbsession.query(Edge).all()  # get all edges from db
+    for tract in tracts:
+        graph.add_node(tract)
+    for edge in edges:
+        source = request.dbsession.query(Tract).get(edge.tract_source)
+        target = request.dbsession.query(Tract).get(edge.tract_target)
+        graph.add_edge(source, target)
+    return graph
+
+
+class OccupiedDist(object):
+    """A stucture to contain and separate tracts in a State object.
+
+    add_node(self, node): adds node to nodes and updates district
+    properties accordingly
+
+    rem_node(self, node): removes node from nodes and updates district
+    properties accordingly
+    """
+
+    def __init__(self, districtID, state_graph, tracts=None):
+        """."""
+        self.nodes = nx.Graph()
+        self.perimeter = []
+        self.population = 0
+        self.area = 0
+        self.districtID = districtID
+        if tracts:
+            try:
+                for tract in tracts:
+                    self.add_node(tract, state_graph)
+            except TypeError:
+                raise TypeError('Tracts must be iterable.')
+
+    def add_node(self, node, state_graph):
+        """Add node to nodes and updates district properties."""
+        node.districtid = self.districtID
+        self.nodes.add_node(node)
+        for edge in state_graph.neighbors(node):
+            if edge in self.nodes.nodes():
+                self.nodes.add_edge(edge, node)
+        self.population += node.tract_pop
+        self.area += node.shape_area
+        if node in self.perimeter:
+            self.perimeter.remove(node)
+        neighbors = state_graph.neighbors(node)
+        for neighbor in neighbors:
+            if neighbor not in self.nodes.nodes() and neighbor not in self.perimeter:
+                self.perimeter.append(neighbor)
+
+    def rem_node(self, node, state_graph):
+        """Remove node from nodes and updates district properties."""
+        self.population -= node.tract_pop
+        self.nodes.remove_node(node)
+        self.area -= node.shape_area
+        neighbors = state_graph.neighbors(node)
+        to_perimeter = False
+        for neighbor in neighbors:
+            takeout = True
+            if neighbor in self.perimeter:
+                neighborneighbors = state_graph.neighbors(neighbor)
+                for neighborneighbor in neighborneighbors:
+                    if neighborneighbor in self.nodes.nodes():
+                        takeout = False
+                if takeout:
+                    self.perimeter.remove(neighbor)
+            elif neighbor in self.nodes.nodes():
+                to_perimeter = True
+        if to_perimeter:
+            self.perimeter.append(node)
+
+
+class UnoccupiedDist(OccupiedDist):
```

Comment:
Below you override every method from the `OccupiedDist` object. Why did you inherit from it instead of just constructing a new object?